erpnext/public/js/controllers/taxes_and_totals.js

```diff
@@ -390,10 +390,21 @@ erpnext.taxes_and_totals = erpnext.payments.extend({
 		this.frm.doc.grand_total = flt(tax_count
 			? this.frm.doc["taxes"][tax_count - 1].total + flt(this.frm.doc.rounding_adjustment)
 			: this.frm.doc.net_total);
-
+		var payment_status = true;
 		if(in_list(["Quotation", "Sales Order", "Delivery Note", "Sales Invoice"], this.frm.doc.doctype)) {
 			this.frm.doc.base_grand_total = (this.frm.doc.total_taxes_and_charges) ?
 				flt(this.frm.doc.grand_total * this.frm.doc.conversion_rate) : this.frm.doc.base_net_total;
+			if (this.frm.doc.is_pos && this.frm.doc.is_return) {
+				$.each(this.frm.doc['payments'] || [], function (index, data) {
+					if (payment_status) {
+						data.amount = me.frm.doc.base_grand_total;
+						data.base_amount = flt(data.amount * me.frm.doc.conversion_rate, precision("base_amount"));
+						payment_status = false;
+					}
+				});
+				this.frm.doc.paid_amount = flt(me.frm.doc.base_grand_total, precision("paid_amount"));
```

Comment:
paid_amount should be in invoice currency

Reply:
Does it mean to use `base_paid_amount` field instead of `paid_amount`?

Reply:
`this.frm.doc.paid_amount = flt(me.frm.doc.grand_total, precision("paid_amount"));`

Reply:
Thanks for feedback Updated.